yb-voyager/src/monitor/monitor_target_health.go

```diff
@@ -0,0 +1,264 @@
+	yb TargetDBForMonitorHealth
+
+	//Skip booleans as per flags
+	skipNodeCheck        bool
+	skipDiskCheck        bool
+	skipReplicationCheck bool
+
+	//Display function for the printing any msg on the console
+	displayMsgFunc func(string)
+
+	tconf tgtdb.TargetConf
+}
+
+func NewMonitorTargetYBHealth(yb TargetDBForMonitorHealth, skipDiskUsageHealthChecks utils.BoolStr, skipReplicationChecks utils.BoolStr, skipNodeHealthChecks utils.BoolStr, displayMsgFunc func(msg string), ybClient YugabyteDBClient, tconf tgtdb.TargetConf) MonitorTargetYBHealth {
+	return MonitorTargetYBHealth{
+		nodesStatus:          make(map[string]bool),
+		yb:                   yb,
+		skipNodeCheck:        bool(skipNodeHealthChecks),
+		skipDiskCheck:        bool(skipDiskUsageHealthChecks),
+		skipReplicationCheck: bool(skipReplicationChecks),
+		displayMsgFunc:       displayMsgFunc,
+		tconf:                tconf,
+	}
+}
+
+func (m *MonitorTargetYBHealth) StartMonitoring() error {
+
+	var err error
+	loadBalancerEnabled, servers, err := m.yb.GetYBServers()
+	if err != nil {
+		return fmt.Errorf("error fetching servers informtion: %v", err)
+	}
+	for _, server := range servers {
+		m.nodesStatus[server.Host] = true
+	}
+
+	if loadBalancerEnabled {
+		m.skipNodeCheck = true
+	}
+
+	//skipping the disk usage monitoring for now, util the metrics function changes are made
+	m.skipDiskCheck = true
+
+	for {
+		err = m.monitorNodesStatusAndAdapt()
+		if err != nil {
+			log.Errorf("error monitoring the node status and adapt: %v", err)
+		}
+
+		err = m.monitorDiskUsageAndAbort()
+		if err != nil {
+			log.Errorf("error monitoring the disk and memory: %v", err)
+		}
+
+		err = m.monitorReplicationOnTarget()
+		if err != nil {
+			log.Errorf("error monitoring the replication: %v", err)
+		}
+		time.Sleep(time.Duration(MONITOR_HEALTH_FREQUENCY_SECONDS) * time.Second)
+	}
+}
+
+/*
+1. if node goes down
+remove its connections from pool and if any current importBatch is running it will fail and automatic retry will take care of using the rest conns
+Put a msg on the console that node went down continuing import witht rest live nodes
+and not monitoring that node until it comes back up so the msg on the console will be printed once only.
+
+2. if node come back
+the new connections will automatically go to that node -- handled via conn-pool
+Put a msg on the console that node came back and now monitoring that node again to see if it goes down again
+
+3. load balancer on the cluster
+No node status checks will happen as we put the load on the load balancer IP and it will take care of this adaptation automatically
+*/
+func (m *MonitorTargetYBHealth) monitorNodesStatusAndAdapt() error {
+	if m.skipNodeCheck {
+		return nil
+	}
+	_, currentNodes, err := m.yb.GetYBServers()
+	if err != nil {
+		return fmt.Errorf("error fetching servers informtion: %v", err)
+	}
+	downNodes := make([]string, 0)
+	upNodes := make([]string, 0)
+
+	for node, up := range m.nodesStatus {
```

Comment:
I'd recommend variable names that are slightly more helpful. How about something like:
`up` -> `previousUp`
`currentUp := lo.ContainsBy(currentNodes, ...`

Also, use either "up" or "live" consistently.